src/proto/console/pointer/mod.rs

```diff
@@ -27,13 +27,18 @@ impl Pointer {
 
     /// Retrieves the pointer device's current state.
     ///
+    /// Will return None if the state has not changed since the last query.
+    ///
     /// # Errors
-    /// - `NotReady` is returned if the state hasn't changed since the last call.
     /// - `DeviceError` if there was an issue with the pointer device.
-    pub fn state(&self) -> Result<PointerState> {
+    pub fn state(&self) -> Result<Option<PointerState>> {
```

Comment:
In a way, this query modifies the state of the pointer protocol, since two successive calls to state() will not return the same result. Knowing that, I would spontaneously change its self parameter to an `&mut self`. Mind if I do that in a follow-up MR?

Reply:
Will be part of an upcoming MR, we can discuss it there.